File: backend/app/utils/text.py
"""Text sanitisation helpers."""

import re

# Match any HTML tag, including attributes and closing tags.
_HTML_TAG_RE = re.compile(r"<[^>]+>", re.DOTALL)

# Collapse multiple whitespace characters into a single space.
_MULTISPACE_RE = re.compile(r"\s+")


def strip_html_tags(html: str | None) -> str:
    """Remove HTML tags and normalise whitespace.

    Returns an empty string for None input so callers can chain `.strip()`
    safely.
    """
    if not html:
        return ""
    text = _HTML_TAG_RE.sub(" ", html)
    text = _MULTISPACE_RE.sub(" ", text)
    return text.strip()

File: mobile/lib/text.ts
/**
 * Strip HTML tags from a string and normalise whitespace.
 * Returns an empty string for null/undefined input.
 */
export function stripHtml(html: string | null | undefined): string {
  if (!html) return '';
  return html
    .replace(/<[^>]+>/g, ' ')
    .replace(/\s+/g, ' ')
    .trim();
}
